pkg/common/config/types.go

```diff
@@ -0,0 +1,21 @@
+package config
+
+type Config struct {
+	Global struct {
+		VCenterIP           string `gcfg:"ip"`
+		ClusterID           string `gcfg:"cluster-id"`
+		ClusterDistribution string `gcfg:"cluster-distribution"`
+		CAFile              string `gcfg:"ca-file"`
+		Thumbprint          string `gcfg:"thumbprint"`
+	}
+	// Virtual Center configurations
+	VirtualCenter map[string]*VirtualCenterConfig
+}
+
+type VirtualCenterConfig struct {
+	User         string `gcfg:"user"`
+	Password     string `gcfg:"password"`
+	VCenterPort  string `gcfg:"port"`
+	InsecureFlag bool   `gcfg:"insecure-flag"`
+	Datacenters  string `gcfg:"datacenters"`
+}
```

Comment:
Can you make sure that all these fields are being used?, we could eliminate things that are not used. From the testing I see that pods are up, which is fine, but you need to perform some backup and restore to ensure you can actually connect to the VC with this new way.
Post the testing in this PR rather than a separate confluence, that way we can track the testing done by seeing the PR.

Reply:
Resolved in latest commit.